libnmstate/nm/dns.py

```diff
@@ -56,37 +55,32 @@ def get_running():
     return dns_state
 
 
-def get_config():
+def get_config(ipv4_profiles, ipv6_profiles):
     dns_conf = {
         DNS.SERVER: [],
         DNS.SEARCH: []
     }
     tmp_dns_confs = []
-    client = nmclient.client()
-    for ac in client.get_active_connections():
-        # NM prefers IPv6 over IPv4 DNS.
-        for ip_profile in (nm_ipv6.get_ip_profile(ac),
-                           nm_ipv4.get_ip_profile(ac)):
-            if not ip_profile:
-                continue
-            if not ip_profile.props.dns and not ip_profile.props.dns_search:
-                continue
-            priority = ip_profile.props.dns_priority
-            if priority == 0:
-                # ^ The dns_priority in 'NetworkManager.conf' is been ignored
-                #   due to the lacking of query function in libnm API.
-                if ac.get_vpn():
-                    priority = DNS_DEFAULT_PRIORITY_VPN
-                else:
-                    priority = DNS_DEFAULT_PRIORITY_OTHER
 
-            tmp_dns_confs.append(
-                {
-                    'server': ip_profile.props.dns,
-                    'priority': priority,
-                    'search': ip_profile.props.dns_search,
-                }
-            )
+    for ac, ip_profile in chain(ipv4_profiles, ipv6_profiles):
```

Comment:
Should prefer ipv6 over IPv4. Please place ipv4_profiles before ipv4.

Reply:
Done